diff --git a/colorer/src/Colorer-library/src/colorer/Region.h b/colorer/src/Colorer-library/src/colorer/Region.h
index 632bc6f..00d697a 100644
--- a/colorer/src/Colorer-library/src/colorer/Region.h
+++ b/colorer/src/Colorer-library/src/colorer/Region.h
@@ -42,9 +42,12 @@ public:
       For example, <code>def:Comment</code> has <code>def:Syntax</code> parent,
       so, some syntax checking can be made with it's content.
   */
+  __attribute__((noinline)) const Region* This() const {
+    return this;
+  }
   bool hasParent(const Region* region) const
   {
-    const Region* elem = this;
+    const Region* elem = This();
     while (elem != nullptr) {
       if (region == elem) {
         return true;
@@ -82,5 +85,3 @@ protected:
 };
 
 #endif
-
-
diff --git a/colorer/src/pcolorer2/pcolorer.cpp b/colorer/src/pcolorer2/pcolorer.cpp
index c038890..14901c6 100644
--- a/colorer/src/pcolorer2/pcolorer.cpp
+++ b/colorer/src/pcolorer2/pcolorer.cpp
@@ -112,20 +112,7 @@ SHAREDSYMBOL void WINAPI SetStartupInfoW(const struct PluginStartupInfo *fei)
   g3::only_change_at_initialization::setLogLevel(INFO, false);
 #endif
 #endif // #if 0
-  try
-  {
-    XMLPlatformUtils::Initialize();
-  }
-  catch(const XMLException &toCatch)
-  {
-    StrX str(toCatch.getMessage());
-#if 0
-    LOG(ERROR) << "Error during Xerces-c Initialization.\n"
-               << "  Exception message:" << str();
-#endif // #if 0
-    std::cerr << "Error during Xerces-c Initialization.\n"
-              << "  Exception message:" << str();
-  }
+  XMLPlatformUtils::Initialize();
 };
 
 /**
